### PR TITLE
[build] [TensorFlow] Studio analytics Library Version update for TF2.6

### PR DESCRIPTION
**Pull request by ahsan-z-khan**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Updating sagemaker-studio-analytics-extension version

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/2.6/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -142,26 +142,25 @@ RUN ${PIP} install --no-cache-dir -U \
...
     "sparkmagic<1" \
     "sagemaker-studio-sparkmagic-lib<1" \
-    "sagemaker-studio-analytics-extension==0.0.2" \
+    "sagemaker-studio-analytics-extension<1" \
     # Let's install TensorFlow separately in the end to avoid
     # the library version to be overwritten
  && ${PIP} install --no-cache-dir -U \
     ${TF_URL} \
-    tensorflow-io==0.21.0 \
+    tensorflow-io \
```
Comment (saimidu) [inline tensorflow/training/docker/2.6/py3/Dockerfile.cpu:163]: Why unpin this version?

File: tensorflow/training/docker/2.6/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -189,7 +188,8 @@ RUN HOME_DIR=/root \
    && chmod +x /usr/local/bin/testOSSCompliance \
    && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
    && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
-   && rm -rf ${HOME_DIR}/oss_compliance*
+   && rm -rf ${HOME_DIR}/oss_compliance* \
+   && /opt/conda/bin/jupyter-kernelspec remove -fy python3
```
Comment (saimidu) [inline tensorflow/training/docker/2.6/py3/Dockerfile.cpu:192]: Please add this as a separate RUN statement after the step that installs sparkmagic.

File: tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu [hunk 3]
```diff
@@ -224,34 +224,33 @@ RUN ${PIP} install --no-cache-dir -U \
...
    "sparkmagic<1" \
    "sagemaker-studio-sparkmagic-lib<1" \
-   "sagemaker-studio-analytics-extension==0.0.2" \
+   "sagemaker-studio-analytics-extension<1" \
    # Let's install TensorFlow separately in the end to avoid
    # the library version to be overwritten
  && ${PIP} install --no-cache-dir -U \
    ${TF_URL} \
-   tensorflow-io==0.21.0 \
+   tensorflow-io \
```
Comment (saimidu) [inline tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu:245]: Why is the tensorflow-io version unpinned?

File: tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu [hunk 4]
```diff
@@ -336,7 +335,8 @@ RUN HOME_DIR=/root \
    && chmod +x /usr/local/bin/testOSSCompliance \
    && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
    && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
-   && rm -rf ${HOME_DIR}/oss_compliance*
+   && rm -rf ${HOME_DIR}/oss_compliance* \
+   && /opt/conda/bin/jupyter-kernelspec remove -fy python3
```
Comment (saimidu) [inline tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu:339]: Please add this as a separate RUN statement after the step that installs sparkmagic.

File: tensorflow/training/docker/2.6/py3/Dockerfile.cpu (hunk 1 above)
Comment (ahsan-z-khan) [reply tensorflow/training/docker/2.6/py3/Dockerfile.cpu:163]: I rebased from master and it got changed. will fix it

File: tensorflow/buildspec.yml [hunk 5]
```diff
@@ -56,7 +56,7 @@ images:
   BuildTensorflowE3GpuCu112Py38TrainingDockerImage:
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:56]: Modify the buildspec to build TF2.6 images

File: tensorflow/training/docker/2.6/py3/Dockerfile.cpu [hunk 6]
```diff
@@ -31,7 +31,7 @@ ARG PIP=pip3
 # disable prompts during package installation
 ARG DEBIAN_FRONTEND=noninteractive
 
-ARG TF_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow/r2.6_aws/cpu/tensorflow-2.6.2-cp38-cp38-manylinux2010_x86_64.whl
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.6_aws/20210816_235124/cpu/py38/tensorflow_cpu-2.6.0-cp38-cp38-manylinux2010_x86_64.whl
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.6/py3/Dockerfile.cpu:34]: We would need the binary for TF2.6.2 instead of TF2.6.0

File: tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu [hunk 7]
```diff
@@ -35,7 +35,7 @@ ARG BRANCH_OFI=1.1.3-aws
 # disable prompts during package installation
 ARG DEBIAN_FRONTEND=noninteractive
 
-ARG TF_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow/r2.6_aws/gpu/tensorflow-2.6.2-cp38-cp38-manylinux2010_x86_64.whl
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.6_aws/20210816_235124/gpu/cu112/py38/tensorflow_gpu-2.6.0-cp38-cp38-manylinux2010_x86_64.whl
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu:38]: We would need the binary for TF2.6.2 instead of TF2.6.0

**Comment by tejaschumbalkar** [comment]
LGTM. lets revert the temp configurations

File: dlc_developer_config.toml [hunk 8]
```diff
@@ -30,7 +30,7 @@ do_build = true
 [test]
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:30]: this file should be reverted

File: tensorflow/buildspec-temp.yml [hunk 9]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec-temp.yml:1]: this file to be reverted

File: tensorflow/buildspec.yml [hunk 10]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:1]: this file to be reverted

**Review by tejaschumbalkar — APPROVED** [review]
